obspy/signal/trigger.py

```diff
@@ -428,6 +428,46 @@ def pk_baer(reltrc, samp_int, tdownmax, tupevent, thr1, thr2, preset_len,
         return pptime.value + 1, pfm.value.decode('utf-8')
 
 
+def aic(td):
+    """
+    Differently from AR-AIC picker, this function calculates AIC function
+    directly from the data, without using the AR coefficients (Maeda, 1985)
+
+    Computes P-phase arrival time digital single-component acceleration
+    or broadband velocity record without requiring threshold settings using
+    AKAIKE INFORMATION CRITERION.
+    Returns P-phase arrival time index and the characteristic function.
+    The returned index correspond to the characteristic function's minima.
+
+    :type td: numpy.ndarray
+    :param td: time series as numpy.ndarray float32 data
+    :return: (idx, aic_cf) idx sample number of parrival; aic_cf numpy.ndarray
+        containing the values of the characteristic function.
+
+    .. seealso:: [Maeda1985]_
+    """
+    # --------------------  Creation of the characteristic function
+    aic_cf = np.zeros(td.size-1)
+    for ii in range(1, td.size):
+        with np.errstate(divide='raise'):
```

Comment:
might be better to put this outside of the for-loop, otherwise numpy might be resetting it's settings twice per loop

Reply:
It is indeed better and computational faster, thanks for the comment